src/gwt/src/org/rstudio/studio/client/workbench/addins/Addins.java

```diff
@@ -73,6 +84,47 @@ public final static RAddin decode(String decoded)
       protected RAddins() {}
    }
    
+   public static class AddinExecutor
+   {
+      public AddinExecutor()
+      {
+      }
+      
+      @Inject
+      private void initialize(AddinsServerOperations server, EventBus events)
+      {
+         server_ = server;
+         events_ = events;
+      }
+      
+      public void execute(RAddin addin)
+      {
+         if (!injected_)
```

Comment:
Note that I use lazy injection here because otherwise these were getting over-eagerly injected. The `AddinsMRUList` also gets its own injected `AddinExecutor` and that was causing injection errors in this class (as the RStudioGinjector was not yet constructed?)